sqlitecluster/SQLiteNode.cpp

```diff
@@ -2580,6 +2541,62 @@ void SQLiteNode::_recvSynchronize(Peer* peer, const SData& message) {
         throw "commits remaining at end";
 }
 
+SQLiteNode::Peer* SQLiteNode::_updateSyncPeer()
+{
+    Peer* newSyncPeer = 0;
+    uint64_t commitCount = _db.getCommitCount();
+    for (auto peer : peerList) {
+        // If any of these conditions are true, then we can't use this peer.
+        if (peer->test("Permaslave") ||                 // Peer is a permaslave
+            !peer->test("LoggedIn") ||                  // Peer is not logged in.
+            !SIEquals((*peer)["State"], "SLAVING") ||   // Peer is not slaving.
+            peer->calcU64("CommitCount") <= commitCount // Peer does not have a greater commit count than us.
+            ) {
+            continue;
+        }
+
+        // Any peer that makes it to here is a usable peer, so it's by default better than nothing.
+        if (!newSyncPeer) {
+            newSyncPeer = peer;
+        }
+        // If the previous best peer and this one have the same latency (meaning they're probably both 0), the best one
+        // is the one with the highest commit count.
+        else if (newSyncPeer->latency == peer->latency) {
+            if (peer->calc64("CommitCount") > newSyncPeer->calc64("CommitCount")) {
+                newSyncPeer = peer;
+            }
+        }
+        // If the existing best has no latency, then this peer is faster (because we just checked if they're equal and
+        // 0 is the slowest latency value).
+        else if (newSyncPeer->latency == 0) {
+            newSyncPeer = peer;
+        }
+        // Finally, if this peer is faster than the best, but not 0 itself, it's the new best.
+        else if (peer->latency != 0 && peer->latency < newSyncPeer->latency) {
+            newSyncPeer = peer;
+        }
+    }
+    
+    // Log that we've changed peers.
+    if (_syncPeer != newSyncPeer) {
+        string from, to;
```

Comment:
I like it when the clauses are simple. When they get long like this I think the `?` and `:` get lost in a sea of other operators and punctuation and make it hard to read.

As an aside, I really like the shortcut format of :
`result = a ?: b`

Which is shorthand for:
`result = a ? a : b`

But nobody seems to have ever heard of it so I never use it because I'm afraid it'll confuse people.